### PR TITLE
feat: add INITIALIZE event to track configuration options

### PR DESCRIPTION
**Pull request by tomiir**
# Description

Adds INITIALIZE event to track configuration options

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/core/src/utils/TypeUtil.ts [hunk 1]
```diff
@@ -18,6 +18,30 @@ import type { ConstantsUtil } from './ConstantsUtil.js'
...
+      enabled?: boolean
+      nonceRefetchIntervalMs?: number
+      sessionRefetchIntervalMs?: number
+      signOutOnDisconnect?: boolean
+      signOutOnAccountChange?: boolean
+      signOutOnNetworkChange?: boolean
+    }
+  }
+  themeMode?: 'dark' | 'light'
+
```
Comment (enesozturk) [inline packages/core/src/utils/TypeUtil.ts:34]: Unexpected empty line maybe?

File: packages/core/src/utils/TypeUtil.ts [hunk 2]
```diff
@@ -18,6 +18,30 @@ import type { ConstantsUtil } from './ConstantsUtil.js'
 import type { ReownName } from '../controllers/EnsController.js'
 import type UniversalProvider from '@walletconnect/universal-provider'
 
+type AppKitConfigs = {
```
Comment (enesozturk) [inline packages/core/src/utils/TypeUtil.ts:21]: Can we give another name specific to event to not confuse with other type definitions? Like `InitializeEventAppKitParams`

**Review by enesozturk — APPROVED** [review]
@tomiir Approved with few comments, also as GH linter shows, there are few console logs 🙌🏽

**Review by lukaisailovic — APPROVED** [review]
@tomiir Can we merge?

**Pull request merged**